diff --git a/ws2812.py b/ws2812.py
index 3db25e0..44feb59 100644
--- a/ws2812.py
+++ b/ws2812.py
@@ -27,10 +27,10 @@ class WS2812:
     """
     buf_bytes = (0x88, 0x8e, 0xe8, 0xee)
 
-    def __init__(self, spi_bus=1, led_count=1, intensity=1):
+    def __init__(self, spi_bus=0, led_count=1, intensity=1, baudrate=3200000):
         """
         Params:
-        * spi_bus = SPI bus ID (1 or 2)
+        * spi_bus = SPI bus ID (0 for W600)
         * led_count = count of LEDs
         * intensity = light intensity (float up to 1)
         """
@@ -42,7 +42,7 @@ class WS2812:
         self.buf = bytearray(self.buf_length)
 
         # SPI init
-        self.spi = pyb.SPI(spi_bus, pyb.SPI.MASTER, baudrate=3200000, polarity=0, phase=1)
+        self.spi = pyb.SPI(spi_bus, baudrate=baudrate, polarity=0, phase=1, sck=pyb.Pin(pyb.Pin.PB_16), mosi=pyb.Pin(pyb.Pin.PB_18), miso=pyb.Pin(pyb.Pin.PB_17), cs=pyb.Pin(pyb.Pin.PB_15))
 
         # turn LEDs off
         self.show([])
@@ -60,7 +60,7 @@ class WS2812:
         """
         Send buffer over SPI.
         """
-        self.spi.send(self.buf)
+        self.spi.write(self.buf)
         gc.collect()
 
     def update_buf(self, data, start=0):
